test/hotspot/jtreg/compiler/c2/irTests/TestContinuationPinningAndEA.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @requires vm.continuations
+ * @bug 8347997
+ * @library /test/lib /
+ * @summary Test that Continuation.pin() and unpin() intrinsics work with EA.
+ * @modules java.base/jdk.internal.vm
+ * @run driver compiler.c2.irTests.TestContinuationPinningAndEA
+ */
+
+package compiler.c2.irTests;
+
+import compiler.lib.ir_framework.*;
+import jdk.internal.vm.Continuation;
+
+public class TestContinuationPinningAndEA {
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Run(test = {
+            "test_FailsEA", "test_Crashes",
+            "test_FailsEANoInline", "test_CrashesNoInline",
+    })
+    public void runMethod() {
+        try {
+            test_FailsEA();
+        } catch (Throwable _) {
```

Comment:
These should normally not throw, right? I would just propagate the exception upwards.

Reply:
And isn't `Exception` or a subclass sufficient?